digits/scheduler.py

```diff
@@ -175,6 +177,30 @@ def add_job(self, job):
             return False
         else:
             self.jobs.append(job)
+
+            if flask._app_ctx_stack.top is not None:
```

Comment:
You're talking about this test?
https://github.com/NVIDIA/DIGITS/blob/v2.2.1/digits/test_scheduler.py#L42-L47

That's a pretty sloppy test - my bad.

You could also give Jobs a default type or make `job_type()` return `None` instead of throwing an error. Both seem easier than subclassing Job to me.